crates/biome_js_analyze/src/analyzers/nursery/no_useless_lone_block_statements.rs

```diff
@@ -0,0 +1,125 @@
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-lone-blocks
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (foo) {
+    ///   bar();
+    ///   {
+    ///     baz();
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// while (foo) {
+    ///   bar();
+    /// }
+    /// ```
+    ///
+    pub(crate) NoUselessLoneBlockStatements {
+        version: "next",
+        name: "noUselessLoneBlockStatements",
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessLoneBlockStatements {
+    type Query = Semantic<JsBlockStatement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let block = ctx.query();
+
+        if let Some(_labeled_statement) = JsLabeledStatement::cast(block.syntax().parent()?) {
+            return None;
+        }
+
+        if block.statements().is_empty() {
+            return Some(block.range());
+        }
+
+        if block_has_control_structure(&block) {
+            return None;
+        }
+
+        for statement in block.statements() {
+            return match statement {
+                AnyJsStatement::JsVariableStatement(variable) if is_let_or_const_statement(&variable) => None,
+                AnyJsStatement::JsClassDeclaration(_) => None,
+                AnyJsStatement::JsFunctionDeclaration(func) if is_block_scoped_function(&func) => None,
+                _ => Some(block.range()),
+            }
+        }
+
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, range: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                range,
+                markup! {
+                    "This block statement doesn't serve any purpose and can be safely removed."
+                },
+            )
+                .note(markup! {
+                "Standalone block statements without any block-level declarations are redundant in JavaScript and can be removed to simplify the code."
+            }),
+        )
+    }
+}
+
+fn block_has_control_structure(block: &JsBlockStatement) -> bool {
+    match block.parent() {
+        Some(AnyJsStatement::JsIfStatement(_)) => true,
+        Some(AnyJsStatement::JsWhileStatement(_)) => true,
+        Some(AnyJsStatement::JsDoWhileStatement(_)) => true,
+        Some(AnyJsStatement::JsForStatement(_)) => true,
+        Some(AnyJsStatement::JsForInStatement(_)) => true,
+        Some(AnyJsStatement::JsForOfStatement(_)) => true,
+        Some(AnyJsStatement::JsWithStatement(_)) => true,
+        Some(AnyJsStatement::JsSwitchStatement(_)) => true,
+        Some(AnyJsStatement::JsTryStatement(_)) => true,
+        _ => false,
+    }
```

Comment:
You can use `matches!()`

```suggestion
    matches!(block.parent(), Some(AnyJsStatement::JsIfStatement(_) | AnyJsStatement::JsWhileStatement(_) | ...))
```

Reply:
The linter suggested this too! Have updated.